src/routes/privateRpc.ts

```diff
@@ -116,4 +117,45 @@ export default function (router: ConnectRouter) {
       throw new ConnectError(error.message, Code.Internal);
     }
   });
+
+  router.rpc(
+    SourceRequestService,
+    SourceRequestService.methods.create,
+    async (req, context) => {
+      if (!context.values.get(baseRpcContext).service) {
+        throw new ConnectError('unauthenticated', Code.Unauthenticated);
+      }
+
+      const originalReq = req.clone();
+      const con = await createOrGetConnection();
+
+      try {
+        req.url = standardizeURL(req.url);
+
+        const sourceRequest = await con.getRepository(SourceRequest).insert({
+          sourceUrl: req.url,
+          userId: 'yggdrasil',
```

Comment:
we have this user?

Reply:
It's not needed as sourceRequest is a raw table (see description)
(SourceRequest is not a linked table so just raw insert works fine :))

Reply:
Oh, never noticed it 😆 I would be baffled why it did not work. Ok, that its good for now, but we could add yggdrasil user as we have ghost either way (not blocking for this PR)

Reply:
I added one first, then thought it would just be spam 😅 (and this user would grift lots of reputation I guess)